### PR TITLE
Add `circleStrokeSpin`, `gear`, `heartBeat`, `triforce` activity indicators

### PR DESCRIPTION
**Pull request by phimage**
Add in demo app, collection view to display all activity indicators
ezgif com-video-to-gif

more idea could be found here https://loading.io/ or from well known iOS applications

### CODE REVIEW

File: IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+//
+//  ActivityIndicatorCollectionViewController.swift.swift
```
Comment (JakeLin) [inline IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift]: The file name seems wrong 🤣

File: IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+//  IBAnimatableApp
+//
+//  Created by phimage on 02/05/2018.
+//  Copyright © 2018 IBAnimatable. All rights reserved.
+//
+
+import UIKit
+import IBAnimatable
+
+final class ActivityIndicatorCollectionViewController: UICollectionViewController {
```
Comment (JakeLin) [inline IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift]: that's awesome to have them all in one place

File: Sources/ActivityIndicators/Animations/ActivityIndicatorAnimationCircleStrokeSpin.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    groupAnimation.animations = [rotationAnimation, strokeEndAnimation, strokeStartAnimation]
+    groupAnimation.duration = strokeStartDuration + beginTime
+    groupAnimation.repeatCount = .infinity
+    groupAnimation.isRemovedOnCompletion = false
+    groupAnimation.fillMode = kCAFillModeForwards
+    return groupAnimation
+  }
+
+  var rotationAnimation: CABasicAnimation {
+    let rotationAnimation = CABasicAnimation(keyPath: "transform.rotation")
```
Comment (JakeLin) [inline Sources/ActivityIndicators/Animations/ActivityIndicatorAnimationCircleStrokeSpin.swift]: I think we can use `CABasicAnimation(keyPath: .rotation)` now

**Review by JakeLin — APPROVED** [review]
Awesome animations, great PR 👍

File: Sources/ActivityIndicators/Animations/ActivityIndicatorAnimationCircleStrokeSpin.swift (hunk 3 above)
Comment (phimage) [reply Sources/ActivityIndicators/Animations/ActivityIndicatorAnimationCircleStrokeSpin.swift]: Yes, I forgot my enum ;(

File: IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift (hunk 1 above)
Comment (phimage) [reply IBAnimatableApp/IBAnimatableApp/Playground/ActivityIndicators/ActivityIndicatorCollectionViewController.swift.swift]: copy paste is bad isn't it

**Comment by phimage** [comment]
I will make the fix, changelog
Maybe add a new animation if I want to code again
Add a compile flag for my easter egg (yes there is one, also one in moon mask type)

and doc
https://github.com/IBAnimatable/IBAnimatable/blob/master/Documentation/ActivityIndicators.md
I am wondering the colors used in background to make the images (I see a gradient but my eyes could failed)

**Comment by phimage** [comment]
new one, instagram inspired
ezgif com-video-to-gif 1

I do not edit the doc yet

**Comment by phimage** [comment]
ezgif com-video-to-gif 3

**Comment by phimage** [comment]
- [ ] Doc : I need to know how to generate animated gif
- [ ] swiftlint warning

**Comment by tbaranes** [comment]
About the gifs, I used a soft recommended by JakeLin: licecap If I remember well. It's pretty simple and straightforward. Also, there's a gradient on it, but I have no suggestion for that, it's one from the library, and I do think I choose it randomly 😬

**Review by tbaranes — APPROVED** [review]

**Comment by JakeLin** [comment]
@phimage those new animations are great 👍. I use licecap for most of the time for simple cases. And I use ScreenFlow for more complicated one like the Gif on README.md. 

@tbaranes The gradient looks awesome on the phone and simulator. But a little over for a Gif since the quality has been decreased a lot. I think we don't have to use the gradient for those new animations, how do you think?

**Comment by tbaranes** [comment]
Fully agree, we can't go with simple flat colors.

**Pull request merged**